nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -210,7 +214,18 @@ public static TreePath findEnclosingMethodOrLambdaOrInitializer(TreePath path) {
     if (annot == null) {
       return null;
     }
-    return AnnotationUtils.getElementValue(annot, "value", String.class, true);
+
+    Map<? extends ExecutableElement, ? extends AnnotationValue> elementValues =
+        annot.getElementValues();
+    for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry :
+        elementValues.entrySet()) {
+      ExecutableElement elem = entry.getKey();
+      if (elem.getSimpleName().contentEquals("value")) {
+        return (String) entry.getValue().getValue();
+      }
+    }
+    // not found
+    return null;
```

Comment:
Was this method removed from `AnnotationUtils` and we are just inlining the implementation here? Or is there a different reason for this change?

Reply:
The relevant methods from `AnnotationUtils` were deprecated.  The replacement APIs in `AnnotationUtils` take an `ExecutableElement` for the annotation rather than an annotation name, I think for efficiency.  We use these APIs from our contract-related handlers, and it would have taken a moderate refactoring (and possibly changes to the handler APIs) to be able to find and keep around the relevant `ExecutableElement`s.  Instead, I re-implemented these methods, under the assumption that if our performance was ok before, this should not cause a regression.

Let me know if you think we should document this better.

Reply:
You are right that the deprecation was because of performance.